leetcode/3.LongestWithoutRepeatingCharacters.cpp

```diff
@@ -0,0 +1,40 @@
+class Solution
+{
+public:
+    // 3. Longest Substring Without Repeating Character: https://leetcode.com/problems/longest-substring-without-repeating-characters/
+
+    // Time Complexity: O(n)
+    // Space Complexity: O(1)
+
+    int lengthOfLongestSubstring(string_view s)
+    {
+        // This problem uses only ASCII characters
+        // Index is used to store indices
+        // of 256 of ASCII characters
+        // Instantiate with -1 as placeholder
+        // Space Complexity: O(1) because it is O(256), which is constant
+        vector<int> indexOfChars(256, -1);
```

Comment:
Cool insight!

Reply:
The coolest engineer telling me I have a cool insight... :swoon: